File: Ruby/1041 - Coordenadas de um ponto.rb
info = gets.split(' ')

x = info[0].to_f
y = info[1].to_f

if y < 0 and x < 0 then
	puts "Q3"
elsif y < 0 and x > 0 then
	puts "Q4"
elsif y > 0 and x > 0 then
	puts "Q1"
elsif y > 0 and x < 0 then
	puts "Q2"
elsif x == 0 and y != 0 then
	puts "Eixo Y"
elsif x != 0 and y == 0 then
	puts "Eixo X"
elsif x == 0 and y == 0 then
	puts "Origem"
end

File: Lua/1041 - Coordenadas de um ponto.lua
function split(info)

	word = ""
	words = {}
	index = 0
	info = info .. " "

	for i=1, string.len(info) do
		cha = string.sub(info, i, i)

		if cha ~= " " then
			word = word .. cha
		else
			if word ~= nil then
				words[index] = word
				index = index + 1
				word = ""
			else
				break
			end
		end
	end

	return words

end


info = io.read()

info = split(info)

x = tonumber(info[0])
y = tonumber(info[1])

if y < 0 and x < 0 then
	print("Q3")
elseif y < 0 and x > 0 then
	print("Q4")
elseif y > 0 and x > 0 then
	print("Q1")
elseif y > 0 and x < 0 then
	print("Q2")
elseif x == 0 and y ~= 0 then
	print("Eixo Y")
elseif x ~= 0 and y == 0 then
	print("Eixo X")
elseif x == 0 and y == 0 then
	print("Origem")
end